proc/groupby.go

```diff
@@ -131,63 +151,73 @@ type GroupByAggregator struct {
 	// lookup table so that values with the same encoding but of
 	// different types do not collide.  No types from this context
 	// are ever referenced.
-	kctx        *resolver.Context
-	keys        []GroupByKey
-	reducerDefs []compile.CompiledReducer
-	builder     *ColumnBuilder
-	// For a regular group-by, tables has one entry with key 0.  For a
-	// time-binned group-by, tables has one entry per bin and is keyed by
-	// bin timestamp (so that a bin with span [ts, ts+timeBinDuration) has
-	// key ts).
-	tables          map[nano.Ts]map[string]*GroupByRow
-	TimeBinDuration int64 // Zero means regular group-by (no time binning).
-	reverse         bool
-	logger          *zap.Logger
-	limit           int
+	kctx         *resolver.Context
+	keys         []GroupByKey
+	reducerDefs  []compile.CompiledReducer
+	builder      *ColumnBuilder
+	table        map[string]*GroupByRow
+	reverse      bool
+	logger       *zap.Logger
+	limit        int
+	valueSortFn  expr.ValueSortFn // to compare primary keys for early key output
+	recordSortFn expr.SortFn
+	maxKey       *zng.Value
+	inputSortDir int
 }
 
 type GroupByRow struct {
 	keycols  []zng.Column
 	keyvals  zcode.Bytes
-	ts       nano.Ts
+	primary  *zng.Value // for sorting when input sorted
```

Comment:
```suggestion
	group  *zng.Value // for sorting when input sorted
```
Or groupval or groupVal or something else.